scripts/generate.js

```diff
@@ -58,23 +59,33 @@ function start (opts) {
       { encoding: 'utf8' }
     )
 
-    const { fn: factory, types } = genFactory(apiOutputFolder, [apiFolder, xPackFolder])
+    const { fn: factory, types, kibanaTypes } = genFactory(apiOutputFolder, [apiFolder, xPackFolder])
     writeFileSync(
       mainOutputFile,
       factory,
       { encoding: 'utf8' }
     )
 
-    const oldTypeDefString = readFileSync(typeDefFile, 'utf8')
-    const start = oldTypeDefString.indexOf('/* GENERATED */')
-    const end = oldTypeDefString.indexOf('/* /GENERATED */')
-    const newTypeDefString = oldTypeDefString.slice(0, start + 15) + '\n' + types + '\n  ' + oldTypeDefString.slice(end)
+    let oldTypeDefString = readFileSync(typeDefFile, 'utf8')
+    let start = oldTypeDefString.indexOf('/* GENERATED */')
+    let end = oldTypeDefString.indexOf('/* /GENERATED */')
+    let newTypeDefString = oldTypeDefString.slice(0, start + 15) + '\n' + types + '\n  ' + oldTypeDefString.slice(end)
```

Comment:
nit: it's easy to forget to update length if we change **start** token. Do you think you could you refer it explicitly?
```js
const startToken = '/* GENERATED */';
let startPosition = oldTypeDefString.indexOf(startToken)
let newTypeDefString = oldTypeDefString.slice(0, start + startToken.length) + ...
```

Reply:
I don't expect it to change, also, we could still forget to update it here :D

Reply:
@delvedor Jumping in on @restrry comment. The recommended change is not only to cover the case where "start token" would change, but also to avoid "magic" number as per our styleguide (https://github.com/elastic/kibana/blob/master/STYLEGUIDE.md#magic-numbersstrings).
When I read `15` it does not mean anything, when I read `startToken.length` it is explicit what it is. Just my 2 cents 😊